deepcell/applications/cytoplasm_segmentation.py

```diff
@@ -0,0 +1,97 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Cytoplasmic segmentation application"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from tensorflow.python.keras.utils.data_utils import get_file
+
+from deepcell_toolbox.deep_watershed import deep_watershed
+
+from deepcell.utils.retinanet_anchor_utils import generate_anchor_params
+from deepcell import model_zoo
+from deepcell.applications import Application
+from deepcell.model_zoo import PanopticNet
+
+
+WEIGHTS_PATH = ('https://deepcell-data.s3-us-west-1.amazonaws.com/'
+                'model-weights/general_cyto_9c7b79e6238d72c14ea8f87023ac3af9.h5')
+
+
+class CytoplasmSegmentation(Application):
+
+    def __init__(self,
+                 use_pretrained_weights=True,
+                 model_image_shape=(128, 128, 1)):
+
+        self.model = PanopticNet('resnet50',
+                                 input_shape=model_image_shape,
+                                 norm_method='whole_image',
+                                 num_semantic_heads=3,
+                                 num_semantic_classes=[1, 1, 2],
+                                 location=True,
+                                 include_top=True)
+
+        if use_pretrained_weights:
+            weights_path = get_file(
+                os.path.basename(WEIGHTS_PATH),
+                WEIGHTS_PATH,
+                cache_subdir='models',
+                md5_hash='4e9136df5071930a66365b2229fc358b'
+            )
+
+            self.model.load_weights(weights_path)
+        else:
+            weights_path = None
+
+        dataset_metadata = {
+            'name': 'general_cyto',
+            'other': 'Pooled phase and fluorescent cytoplasm data - computationally curated'
+        }
+
+        model_metadata = {
+            'batch_size': 2,
+            'lr': 1e-5,
+            'lr_decay': 0.95,
+            'training_seed': 0,
+            'n_epochs': 8,
+            'training_steps_per_epoch': 7899 // 2,
+            'validation_steps_per_epoch': 1973 // 2
+        }
+
+        super(CytoplasmSegmentation, self).__init__(self.model,
+                                                    model_image_shape=model_image_shape,
+                                                    model_mpp=0.65,
+                                                    preprocessing_fn=None,
+                                                    postprocessing_fn=deep_watershed,
+                                                    dataset_metadata=dataset_metadata,
+                                                    model_metadata=model_metadata)
+
+        self.predict = self._predict_segmentation
```

Comment:
I think it would be clearer to redefine `predict` than to assign it in the init.